MIGRATING.md

```diff
@@ -0,0 +1,71 @@
+   Signature changed to `getLogs(Date since, Date until, String filter, String q, String sortOrder)`
+   
+- `com.okta.sdk.resource.group.GroupList` 
+   - `listGroups(String q, String filter, String expand)`
+   Signature changed to `listGroups(String q, String filter)`
+
+- `com.okta.sdk.resource.application.SwaThreeFieldApplicationSettingsApplication`
+   - `getTargetUrl` Renamed to `getTargetURL`
+   - `setTargetUrl` Renamed to `setTargetURL`
+
+- `com.okta.sdk.resource.group.Group`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.group.GroupType`
+   
+- `com.okta.sdk.resource.group.rule.GroupRule`
+   - `delete(Boolean removeUsers)` Signature changed to `delete()`
+   - Removed `getAllGroupsValid` and `setAllGroupsValid` (property `allGroupsValid` is removed)
+   - Removed `getEmbedded` (property `_embedded` is removed)
+
+- `com.okta.sdk.resource.log.LogAuthenticationContext`
+   - `getCredentialProvider` Return type changed from `List` to `com.okta.sdk.resource.log.LogCredentialProvider`
+   - `getCredentialType` Return type changed from `List` to `com.okta.sdk.resource.log.LogCredentialType`
+
+- `com.okta.sdk.resource.policy.PolicyRule`
+   - Removed `setId` (property `id` is read-only)
+
+- `com.okta.sdk.resource.user.factor.VerifyFactorRequest`
+   - Removed `getTokenLifetimeSeconds` & `setTokenLifetimeSeconds` (property `tokenLifetimeSeconds` is removed)
+
+- `com.okta.sdk.resource.user.Role`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.role.RoleType`
+   - `setType` Param type changed from `String` to `com.okta.sdk.resource.role.RoleType`
+
+- All `*Factor*` classes Renamed to `*UserFactor*`
+
+- `com.okta.sdk.resource.user.User`
+   - Renamed `addGroupTarget` to `addGroupTargetToRole`
+   - Renamed `addRole` to `assignRole`
+   - Renamed `listGroupTargetsForRole` to `listGroupTargets`
+   - Renamed `removeGroupTargetFromRole` to `removeGroupTarget`
+   - Renamed `addFactor` to `enrollFactor`
+   - Renamed `listRoles`to `listAssignedRoles`
+   - Removed `forgotPassword` (use `resetPassword` instead)
+   - Removed `endAllSessions` (use `clearSessions` instead)
+   - `listSupportedFactors` Return type changed from `com.okta.sdk.resource.user.factor.FactorList` to `com.okta.sdk.resource.user.factor.UserFactorList`
+   - `expirePassword` Return type changed from `com.okta.sdk.resource.user.TempPassword` to `com.okta.sdk.resource.user.User`
+   - `getFactor` Return type changed from `com.okta.sdk.resource.user.factor.Factor` to `com.okta.sdk.resource.user.factor.UserFactor`
+   - `listFactors` Return type changed from `com.okta.sdk.resource.user.factor.FactorList` to `com.okta.sdk.resource.user.factor.UserFactorList` 
+   - `resetPassword(String provider, Boolean sendEmail)` Signature changed to `resetPassword(Boolean sendEmail)`
+
+- `com.okta.sdk.resource.user.UserCredentials`
+   - Removed `getEmails` & `setEmails` () (property `emails` is removed)
```

Comment:
looks like there is an extra `()` ?
```suggestion
   - Removed `getEmails` & `setEmails` (property `emails` was removed)
```
was the spec previously invalid? (if so maybe just say "this value was always null") or something to show that use does not need to worry about this

Reply:
Looking at https://developer.okta.com/docs/reference/api/users/#example, I see that `emails` is not included in `credentials` (`UserCredentials` model). However, our `v1.6.0` spec has it. I see the revision `v2.0.0` as more inline to the developer documentation here. Having said that, I am not sure how `emails` was used by the Okta core back-end hitherto. I would try to add a helpful note to the reader here.